src/app/api/activity-logs/services/activity-log.service.ts

```diff
@@ -28,49 +31,88 @@ export class ActivityLogService extends BaseService {
     `
     const parsedActivityLogs = DBActivityLogArraySchema.parse(activityLogs)
 
-    // @todo fetch all users - internal as well as client and pass them to each transformer for name, profile picture etc.
+    const copilotService = new CopilotAPI(this.user.token)
+
+    const promises_getInternalUser = parsedActivityLogs.map(async (activityLog) => {
+      return copilotService.getInternalUser(activityLog.userId)
+    })
+
+    const internalUsers = await Promise.all(promises_getInternalUser)
```

Comment:
can't "Client" be initiator @aatbip? when client comments/ changes description etc.

Reply:
Refactored in 9eb591bfaf10b820bcc6cb889d7ea33fad2986b9 commit